depends/gen_id

```diff
@@ -58,6 +58,17 @@
     echo "CXX_STANDARD=${CXX_STANDARD}"
     echo "END CXX"
 
+    # We use lld when cross-compiling for macOS, and it's version should
+    # be tied to LLVM. However someone compiling with GCC and -fuse-ld=lld
+    # would not see a cache bust if the LLVM toolchain was updated.
+    echo "BEGIN lld"
+    bash -c "ld.lld --version"
```

Comment:
I wonder if this could be replaced with an actual invocation of the linker instead? Something like:

`bash -c "echo \"int main(){}\" | $CC -v $CFLAGS $LDFLAGS -Wl,-v -xc -o /dev/null -"`

Reply:
We can do something more involved if you like, also want to avoid anything that is likely to be flaky.

Reply:
Eh, probably not worth bothering with.

Reply:
Mmm, it is a shame that an update to an unused linker would bust the cache. But I guess that'd be so rare that it's not worth being too concerned about.

We can always revisit and switch to an invocation if it turns out to be too greedy.